From 9fcc636dafac734441fcb85241b51ef22c1ca3fb Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Fri, 19 Apr 2024 19:04:14 +0300
Subject: [PATCH] nl80211: Restore libnl3-route inclusion for full VLAN support
 with netlink

The changes in nl80211 to get rid of the libnl3-route dependency are not
sufficient to fully remove the depency from other parts of the code.
Revert the makefile related changes from that commit to avoid build
issues for cases where CONFIG_FULL_DYNAMIC_VLAN=y and
CONFIG_VLAN_NETLINK=y are used without CONFIG_DRIVER_MACSEC_LINUX=y
pulling in the needed library.

Fixes: a210fdb1c717 ("nl80211: Rewrite neigh code to not depend on libnl3-route")
Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 src/drivers/drivers.mak | 1 +
 src/drivers/drivers.mk  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/drivers/drivers.mak b/src/drivers/drivers.mak
index 6b452c64e..a03d4a034 100644
--- a/src/drivers/drivers.mak
+++ b/src/drivers/drivers.mak
@@ -156,6 +156,7 @@ ifdef CONFIG_FULL_DYNAMIC_VLAN
 NEED_LINUX_IOCTL=y
 ifdef CONFIG_VLAN_NETLINK
 NEED_LIBNL=y
+CONFIG_LIBNL3_ROUTE=y
 endif
 endif
 
diff --git a/src/drivers/drivers.mk b/src/drivers/drivers.mk
index de4cb99de..10eab6a92 100644
--- a/src/drivers/drivers.mk
+++ b/src/drivers/drivers.mk
@@ -151,6 +151,7 @@ ifdef CONFIG_FULL_DYNAMIC_VLAN
 NEED_LINUX_IOCTL=y
 ifdef CONFIG_VLAN_NETLINK
 NEED_LIBNL=y
+CONFIG_LIBNL3_ROUTE=y
 endif
 endif
 
-- 
2.43.0

